src/days-2-4-adv/adv.py

```diff
@@ -49,3 +61,86 @@
 # Print an error message if the movement isn't allowed.
 #
 # If the user enters "q", quit the game.
+
+
+def DirectionEval(direction):
+    if hasattr(currentPlayer.room, directionFunctions[direction]):
+        currentPlayer.room = getattr(
+            currentPlayer.room, directionFunctions[direction])
+        return False
+    else:
+        print("Invalid Direction")
+        return True
+
+
+def ItemEval(item, itemsList):
+    return next((y for y in itemsList if y.name.lower() == item), None)
+
+
+def TextEval(text):
+    print(text)
+    if text == "q" or text == "quit":
+        raise AdventureDone
+    elif text in directionFunctions.keys():
+        return DirectionEval(text)
+    elif "score" in text:
+        print(currentPlayer.score)
+        return True
+    elif "grab" in text or "take" in text:
+        if not currentPlayer.room.is_light or not currentPlayer.hasLight:
+            print("Good luck finding that in the dark!")
+            return True
+        itemObj = ItemEval(text[5:], currentPlayer.room.items)
+        if itemObj is not None:
+            itemObj.onTake(currentPlayer)
+            return False
+        print('That object can not be grabbed')
+        return True
+    elif "drop" in text:
+        itemObj = ItemEval(text[5:], currentPlayer.items)
+        if itemObj is not None:
+            itemObj.onDrop(currentPlayer)
+            return False
+        print('That object can not be dropped')
+        return True
+    elif text == "h" or text == "help":
+        print("Valid commands are: (h)elp, (q)uit (n)orth, (w)est, (s)outh, (e)ast, (i)nventory, score, grab, and drop")
+        return True
+    elif text == "i" or text == "inventory":
+        if len(currentPlayer.items) > 0:
+            itemSTR = "Your inventory contains:"
+            for item in currentPlayer.items:
+                itemSTR = itemSTR+" " + item.name
+            print(itemSTR)
+        else:
+            print("Your inventory is empty!")
+        return True
+    else:
+        print("Invalid Direction")
+        return True
+
+
+print(
+    f"Welcome {currentPlayer.name}, pick a direction,use h for help, or use q to quit")
```

Comment:
nitpicky formatting suggestion - put a colon and space, or at least just a space before the end quote. This makes the display a little neater.